CacheSim/GenericHashTable.h

```diff
@@ -314,7 +378,7 @@ class GenericHashTable
     const size_t new_capacity = old_capacity ? old_capacity * 2 : 64;
 
     Elem** old_table = m_Table;
-    Elem** new_table = (Elem**) HeapAlloc(m_Heap, 0, sizeof(Elem*) * new_capacity);
+    Elem** new_table = (Elem**) VirtualMemoryAlloc(sizeof(Elem*) * new_capacity);
```

Comment:
One concrete idea to resolve this would be to grab upstream TLSF, perhaps as a submodule, and to use `mmap`/`VirtualAlloc` to grab some base memory for it. TLSF 3 has support for extending the pool with additional core memory.